File: 프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    answer = []
    seen = set()

    groups = s[2:-2].split('},{')
    groups.sort(key=lambda g: len(g.split(',')))

    for group in groups:
        for v in group.split(','):
            if v not in seen:
                seen.add(v)
                answer.append(int(v))

    return answer

File: 프로그래머스/2/64065. 튜플/튜플.java
import java.util.*;

class Solution {
    
    public int[] solution(String s) {
        String[] groups = s.substring(2, s.length() - 2)
                           .split("\\},\\{");

        Arrays.sort(groups, Comparator.comparingInt(
                group -> group.split(",").length
        ));

        Set<Integer> seen = new HashSet<>();
        List<Integer> answer = new ArrayList<>();

        for (String group : groups) {
            for (String value : group.split(",")) {
                int number = Integer.parseInt(value);

                if (seen.add(number)) {
                    answer.add(number);
                }
            }
        }

        return answer.stream()
                     .mapToInt(Integer::intValue)
                     .toArray();
    }
}